### ISSUE TITLE
omprog plugin: argv[0] not set when using binary without arguments

### ISSUE THREAD

**Issue by hrak** [rsyslog/rsyslog#1858]
When using the omprog plugin with a binary without arguments, argv[0] (the name of the binary) is not set, causing binaries that depend on this value being set to crash or misbehave. 

In my case this is a Go binary using the `flags` library. This library initializes with `var CommandLine = NewFlagSet(os.Args[0], ExitOnError)`, and this causes the program to crash with:

```
panic: runtime error: index out of range

goroutine 1 [running]:
flag.init()
	/usr/local/opt/go/libexec/src/flag/flag.go:952 +0x137
```

My config:

```
module(load="omprog")
module(
  load="impstats"
  interval="10"
  format="json"
  resetCounters="off"
  ruleset="process_stats"
)

ruleset(name="process_stats") {
  action(
    type="omprog"
    name="to_exporter"
    binary="/usr/local/bin/rsyslog_exporter"
  )
}
```

When i changed this config to pass a parameter to the binary (`binary="/usr/local/bin/rsyslog_exporter -web.listen-address :9104"`), the problem disappeared.

**Comment by rgerhards** [comment]
sorry, I will probably not make it into the 8.31 timeframe. To do this right, a bit of good refactoring is needed. The code is hard to maintain at the spot in question. Probably that was the root cause of the issue...

**Comment by rgerhards** [comment]
yeah... as there is a relatively easy work-around and there is some higher regression potential for this fix, I have decided to move it over to 8.32. It's not worth the risk - sorry for that.
